src/sentry/plugins/providers/base.py

```diff
@@ -6,13 +6,66 @@
 from rest_framework.response import Response
 from social_auth.models import UserSocialAuth
 
+from sentry.models import Integration, OrganizationIntegration
+
 from sentry.exceptions import InvalidIdentity, PluginError
 
 
 class ProviderMixin(object):
     auth_provider = None
     logger = None
 
+    def link_auth(self, user, organization, data):
```

Comment:
I figured that not using social auth was probably the exception to the rule, so plugins could override if necessary. this is what i did for the github apps integration: https://github.com/getsentry/sentry-plugins/pull/188/files#diff-038ed6e2271ba416bd35027ef4cf762eR409